app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt

```diff
@@ -169,31 +171,33 @@ class RecordAudioViewModel @Inject constructor(
     }
 
     fun stopRecording() {
-        if (state.buttonState == RecordAudioButtonState.RECORDING) {
-            audioMediaRecorder.stop()
-        }
-        audioMediaRecorder.release()
+        viewModelScope.launch(dispatchers.default()) {
```

Comment:
can we keep the `viewModelScope.launch` as it is and switch dispatchers for generateAudioFileWithEffects here

https://github.com/wireapp/wire-android/blob/7b1afb541948d8761b22e871bb03cc9354e2ec4d/app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/GenerateAudioFileWithEffectsUseCase.kt#L39

Reply:
It's not only for `generateAudioFileWithEffects`, we need also to run `getPlayableAudioFile()`, and `getAudioLengthInMs()`and ` stop()` in worker thread

Reply:
Then we can change of them individual, updating the state from a background thread can cause some issues.

Happy to talk about this in a 1:1 or in the next collective